aggregator/src/aggregator.rs

```diff
@@ -2751,375 +2647,481 @@ where
 /// See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age
 const CORS_PREFLIGHT_CACHE_AGE: u32 = 24 * 60 * 60;
 
-/// Constructs a Warp filter that matches the request path for tasks, as well as whatever regex is
-/// provided, but without consuming the path or any values encoded into it, such that later filters
-/// can still extract path components.
-fn path_matching_filter(trailing_fragment_regex: &str) -> Result<BoxedFilter<()>, Error> {
-    let regex = Regex::new(&format!(
-        // Task ID is 32 bytes, hence 43 bytes of Base64URL
-        "tasks/[A-Za-z0-9-_]{{43}}/{trailing_fragment_regex}"
-    ))
-    .map_err(|e| Error::Internal(format!("error compiling regex: {e}")))?;
+/// Wrapper around a type that implements [`Encode`]. It acts as a Trillium handler, encoding the
+/// inner object and sending it as the response body, setting the Content-Type header to the
+/// provided media type, and setting the status to 200.
+struct EncodedBody<T> {
+    object: T,
+    media_type: &'static str,
+}
 
-    let filter = warp::path::peek()
-        .and_then(move |peek: warp::path::Peek| {
-            let regex = regex.clone();
-            async move {
-                if regex.is_match(peek.as_str()) {
-                    Ok(())
-                } else {
-                    Err(warp::reject::reject())
-                }
-            }
-        })
-        .untuple_one()
-        .boxed();
+impl<T> EncodedBody<T>
+where
+    T: Encode,
+{
+    fn new(object: T, media_type: &'static str) -> Self {
+        Self { object, media_type }
+    }
+}
 
-    Ok(filter)
+#[async_trait]
+impl<T> Handler for EncodedBody<T>
+where
+    T: Encode + Sync + Send + 'static,
+{
+    async fn run(&self, mut conn: Conn) -> Conn {
+        conn.headers_mut()
+            .insert(KnownHeaderName::ContentType, self.media_type);
```

Comment:
Out of curiosity, is this explicit `insert` (vs `with_header`) because it was appending and that wasn't the desired behavior? Asking because I've been wondering if `with_header` should insert by default

Reply:
Ah, chalk this one up to path dependence. That behavior is not an issue here, because the different places content-type are set are exclusive.

Reply:
I just double checked and it actually *does* insert — I already addressed that. I think it's identical to `conn.with_header(KnownHeaderName::ContentType, self.media_type)` here